### ISSUE TITLE
Provide support for all primitive types and arrays for annotations in IDL

### ISSUE THREAD

**Issue by vpopovpcore** [Pelagicore/qface#27]
Currently only strings are supported, there's no way to define:
@config(zones=[left, right, rear]

**Comment by jryannel** [comment]
This could either be achieved by supporting other types (e.g. bool, int, real, string) and arrays and dicts or by specifying that the string given is interpreted as a JSON string and we use python JSON converted todo the job.

```js
@config(zones=["left","right","rear"])
```

**Comment by jryannel** [comment]
The other option would be to use inline YAML syntax.

```js
@config(zones: [left, right, back], hello:{a: 10, b: 20})
```

And treat everything what can appear between the `()` as YAML object (so replace `()` with `{}`) and feet it to the yaml parser. By this we can use a powerful parser for this code and do not have the overhead of all the JSON `"`. It would also be aligned with the YAML external annotation files.

The only issue here is to be able to detect when the YAML part would end.

**Comment by mkrus** [comment]
would be very useful, at least require full support for strings and boolean

**Comment by jryannel** [comment]
@mkrus , @vpopovpcore 
Please have a look here: https://github.com/Pelagicore/qface/tree/feature/yaml_annotations.

The annotation syntax has changed to be YAML based. Same as the external annotations file.

I would like to make you aware of the changes before I merge it in, as it will require to update your annotations.

**Comment by mkrus** [comment]
looks good to me. maybe point our they should be single line in the documentation

**Comment by jryannel** [comment]
This is fixed with c82d515e541c31019784e884e50d093fb955d49a
